tomviz/Module.h

```diff
@@ -127,6 +127,11 @@ public slots:
   /// properties to the panel.
   virtual void addToPanel(QWidget* panel);
 
+  /// This method is called just prior to removing this Module's properties from
+  /// the panel. Subclasses can use it to perform any necessary cleanup such as
+  /// disconnecting some signals and slots.
+  virtual void prepareToRemoveFromPanel(QWidget* panel);
+
```

Comment:
I thought about going that route, making this a signal that is emitted prior to the removal of the `Module` items from the panel, but I'd still need a slot to connect it to. No reason not to, and that might be more generally useful, but some other PR...